web/template.go

```diff
@@ -215,74 +215,101 @@ func tmplChannelOpts(channelTypes []discordgo.ChannelType) interface{} {
 	}
 }
 
-func tmplChannelOptsMulti(channelTypes []discordgo.ChannelType) func(channels []dstate.ChannelState, selections []int64) template.HTML {
+func tmplChannelOptsMulti(allowedChannelTypes []discordgo.ChannelType) func(channels []dstate.ChannelState, selections []int64) template.HTML {
 	return func(channels []dstate.ChannelState, selections []int64) template.HTML {
+		gen := &channelOptsHTMLGenState{allowedChannelTypes: allowedChannelTypes, channels: channels, selections: selections}
+		return gen.HTML()
+	}
+}
 
-		var builder strings.Builder
+type channelOptsHTMLGenState struct {
```

Comment:
Open to suggestions for better names here — channelOptsHTMLGenState is quite a mouthful 😄